rust/protocol/src/error.rs

```diff
@@ -5,186 +5,120 @@
 
 use crate::curve::KeyType;
 
-use std::error::Error;
+use displaydoc::Display;
+use thiserror::Error;
+
 use std::fmt;
 use std::panic::UnwindSafe;
 
 pub type Result<T> = std::result::Result<T, SignalProtocolError>;
 
+/// Wraps a boxed error struct and delegates the [std::error::Error] trait to it.
+///
+/// A [Box] wrapping an error apparently does not implement [std::error::Error] itself, which breaks
```

Comment:
Hm, [that implementation is in the stdlib](https://doc.rust-lang.org/std/error/trait.Error.html#impl-Error-39). Something else must be going on.

Reply:
Unfortunately, there is a distinction between `Box<E: Error>` and `Box<dyn Error>`. I have added this explanatory link to the docstring: https://stackoverflow.com/a/62452516/2518889.

Reply:
I believe we could work around this by adding a type parameter `E: Error` to the definition of `SignalProtocolError`, but that seems more annoying than wrapping in `CallbackErrorWrapper`.